mahta/JS/JS data fetching/script.js

```diff
@@ -0,0 +1,174 @@
+      deleteButton.textContent = "Delete";
+
+      deleteButton.addEventListener("click", () => {
+        const confirmed = confirm("Are you sure you want to delete this post?");
+
+        if (confirmed) {
+          fetch(`https://jsonplaceholder.typicode.com/posts/${post.id}`, {
+            method: "DELETE",
+          })
+            .then(() => {
+              postContainer.remove();
+            })
+            .catch((error) => {
+              console.error(error);
+              body.textContent = "Error deleting post";
+            });
+        }
+      });
+
+      const buttonContainer = document.createElement("div");
+      buttonContainer.classList.add("post-buttons");
+      buttonContainer.appendChild(updateButton);
+      buttonContainer.appendChild(deleteButton);
+
+      postContainer.appendChild(title);
+      postContainer.appendChild(body);
+      postContainer.appendChild(author);
+      postContainer.appendChild(buttonContainer);
+
+      postsContainer.appendChild(postContainer);
+    });
+  })
+  .catch((error) => {
+    console.error(error);
+  });
+
+function redirectToNewPostPage() {
+  window.location.href = "new-post.html";
+}
+
+function submitPost() {
+  const post = {
+    title: document.getElementById("postTitleInput").value,
+    body: document.getElementById("postBodyInput").value,
+  };
+
+  fetch("https://jsonplaceholder.typicode.com/posts", {
+    method: "POST",
+    body: JSON.stringify(post),
+    headers: {
+      "Content-Type": "application/json",
+    },
+  })
+    .then((response) => {
+      if (response.ok) {
+        window.location.href = "index.html";
+      } else {
+        throw new Error("Post submission failed");
+      }
+    })
+    .catch((error) => {
+      console.log(error);
+    });
+}
+function performSearch() {
+  const searchInput = document.getElementById("searchInput").value.trim();
+  if (!searchInput) {
+    getAllPosts();
+    return;
+  }
+
+  fetch(`https://jsonplaceholder.typicode.com/posts?title=${searchInput}`)
+    .then((response) => response.json())
+    .then((data) => {
+      const postsContainer = document.getElementById("posts-container");
+      postsContainer.innerHTML = "";
+
+      if (data.length === 0) {
+        postsContainer.innerHTML = "No matching posts found.";
+        return;
+      }
+
+      data.forEach((post) => {
+        const postElement = createPostElement(post);
+        postsContainer.appendChild(postElement);
+      });
+    })
+    .catch((error) => {
+      console.log(error);
+    });
+}
+
+function getAllPosts() {
+  fetch("https://jsonplaceholder.typicode.com/posts")
+    .then((response) => response.json())
+    .then((data) => {
+      const postsContainer = document.getElementById("posts-container");
+      postsContainer.innerHTML = "";
+
+      data.forEach((post) => {
+        const postElement = createPostElement(post);
+        postsContainer.appendChild(postElement);
+      });
```

Comment:
Please create 10 new posts.

Reply:
? I didn't get what you mean

Reply:
When the `getAllPost()` function is called, it requests 100 posts from the server, but we only need 10.

> The index page shows 10 posts with their title, body, and author name.

the function is used to generate posts on the index page. isn't it? (Line: 129)

@MahtaMirzaei

Reply:
I think it's solved.. but I'm not sure if I must change the part were i use slice for the code.